services/api/optimistic_test.go

```diff
@@ -304,92 +303,94 @@ func TestPrepareBuildersForSlot(t *testing.T) {
 	require.Zero(t, entry.collateral.Cmp(big.NewInt(int64(collateral))))
 }
 
-func TestBuilderApiSubmitNewBlockOptimistic(t *testing.T) {
-	testCases := []struct {
-		description     string
-		wantStatus      common.BuilderStatus
-		simulationError error
-		expectDemotion  bool
-		httpCode        uint64
-		blockValue      uint64
-	}{
-		{
-			description: "success_value_less_than_collateral",
-			wantStatus: common.BuilderStatus{
-				IsOptimistic: true,
-				IsHighPrio:   true,
-			},
-			simulationError: nil,
-			expectDemotion:  false,
-			httpCode:        200, // success
-			blockValue:      collateral - 1,
-		},
-		{
-			description: "success_value_greater_than_collateral",
-			wantStatus: common.BuilderStatus{
-				IsOptimistic: true,
-				IsHighPrio:   true,
-			},
-			simulationError: nil,
-			expectDemotion:  false,
-			httpCode:        200, // success
-			blockValue:      collateral + 1,
-		},
-		{
-			description: "failure_value_more_than_collateral",
-			wantStatus: common.BuilderStatus{
-				IsOptimistic: true,
-				IsHighPrio:   true,
-			},
-			simulationError: errFake,
-			expectDemotion:  false,
-			httpCode:        400, // failure (in pessimistic mode, block sim failure happens in response path)
-			blockValue:      collateral + 1,
-		},
-	}
-
-	for _, tc := range testCases {
-		t.Run(tc.description, func(t *testing.T) {
-			pubkey, secretkey, backend := startTestBackend(t)
-			backend.relay.optimisticSlot.Store(slot)
-			backend.relay.capellaEpoch = 1
-			var randaoHash boostTypes.Hash
-			err := randaoHash.FromSlice([]byte(randao))
-			require.NoError(t, err)
-			withRoot, err := ComputeWithdrawalsRoot([]*consensuscapella.Withdrawal{})
-			require.NoError(t, err)
-			backend.relay.payloadAttributes[emptyHash] = payloadAttributesHelper{
-				slot:            slot,
-				withdrawalsRoot: withRoot,
-				payloadAttributes: beaconclient.PayloadAttributes{
-					PrevRandao: randaoHash.String(),
-				},
-			}
-			pkStr := pubkey.String()
-			rr := runOptimisticBlockSubmission(t, blockRequestOpts{
-				secretkey:  secretkey,
-				pubkey:     *pubkey,
-				blockValue: tc.blockValue,
-				domain:     backend.relay.opts.EthNetDetails.DomainBuilder,
-			}, tc.simulationError, backend)
-
-			// Check http code.
-			require.Equal(t, uint64(rr.Code), tc.httpCode)
-
-			// Check status in db.
-			builder, err := backend.relay.db.GetBlockBuilderByPubkey(pkStr)
-			require.NoError(t, err)
-			require.Equal(t, tc.wantStatus.IsOptimistic, builder.IsOptimistic)
-			require.Equal(t, tc.wantStatus.IsHighPrio, builder.IsHighPrio)
-
-			// Check demotion status is set to expected and refund is false.
-			mockDB, ok := backend.relay.db.(*database.MockDB)
-			require.True(t, ok)
-			require.Equal(t, mockDB.Demotions[pkStr], tc.expectDemotion)
-			require.False(t, mockDB.Refunds[pkStr])
-		})
-	}
-}
+// TODO - bchain these tests fail when we don't consider cancellations. debug them
```

Comment:
did we decide we should modify the handleSubmitBlock function rather then defining a new endpoint? it seems like builders might want to submit different blocks to the RoB endpoint than the full endpoint... but not too sure